scormxblock/scormxblock.py

```diff
@@ -100,23 +100,32 @@
 ]
 
 
-@task(name=u'scormxblock.scormxblock.s3_upload', routing_key=settings.HIGH_PRIORITY_QUEUE)
+@task(name='scormxblock.scormxblock.s3_upload', routing_key=settings.HIGH_PRIORITY_QUEUE)
 def s3_upload(all_content, temp_directory, dest_dir):
     """
     Actual handling of the s3 uploads.
-    It uses a direct boto connection for performance reasons
     """
-    conn = boto.connect_s3(settings.DJFS.get('aws_access_key_id'), settings.DJFS.get('aws_secret_access_key'))
-    bucket = conn.get_bucket(settings.DJFS.get('bucket'))
+    session = boto3.Session(
+        aws_access_key_id=settings.DJFS.get('aws_access_key_id'),
+        aws_secret_access_key=settings.DJFS.get('aws_secret_access_key'),
+        region_name=settings.DJFS.get('region_name'),
+    )
+    s3_client = session.resource('s3')
+    bucket = s3_client.Bucket(settings.DJFS.get('bucket'))
 
     for filepath in all_content:
         sourcepath = path.normpath(path.join(temp_directory.root_path, filepath))
         destpath = path.normpath(path.join(dest_dir, filepath))
-
-        k = boto.s3.key.Key(bucket)
-        k.key = destpath
-        k.set_contents_from_filename(sourcepath)
-        # k.set_acl('public-read')  # Slows calls drastically
+        content_type = mimetypes.guess_type(sourcepath)[0]
```

Comment:
done

Reply:
What do you mean by safe checker

Reply:
Something to check if index 0 is set and avoid raising an exception. @luismorenolopera

Reply:
@Squirrel18  I think it could be left like this, because in all cases it returns a tuple with two elements, let me know if you still want me to make that change

Reply:
I prefer to always have a way to check it, but if you are sure that in all cases it will return a tuple, leave it as is. @luismorenolopera